src/streaming/StreamProcessor.js

```diff
@@ -296,7 +400,7 @@ function StreamProcessor(config) {
     }
 
     function setMediaSource(mediaSource) {
-        bufferController.setMediaSource(mediaSource, getMediaInfo());
+        bufferController.setMediaSource(mediaSource, getMediaInfoArr());
```

Comment:
I suggest to rename the variable in  BufferController.setMediaSource(value,mediaInfo) to mediaInfoArr as well

Reply:
Additional suggestion: Use mediaInfoArr instead of calling the getter function